src/io/jenkins/infra/DockerConfig.groovy

```diff
@@ -0,0 +1,55 @@
+package io.jenkins.infra
+
+
+import java.text.SimpleDateFormat
+import java.util.Date
+
+class DockerConfig {
+  String imageName
+
+  String registry
+
+  String dockerfile
+
+  String credentials
+
+  String mainBranch
+
+  def buildDate
+
+  def infraConfig
+
+  public DockerConfig(String imageName, Map config=[:], InfraConfig infraConfig) {
+    this.imageName = imageName
+
+    this.infraConfig = infraConfig
+
+    this.registry = config.registry
+
+    this.dockerfile = config.dockerfile
+    if (!config.dockerfile) {
+      this.dockerfile = "Dockerfile"
+    }
+
+    this.credentials = config.credentials
+    if (!config.credentials) {
+      this.credentials = "jenkins-dockerhub"
+    }
+
+    this.mainBranch = config.mainBranch
+    if (!config.mainBranch) {
+      this.mainBranch = "master"
+    }
```

Comment:
Cosmetic suggestions to use `map.get('key', 'default')`

```suggestion
    this.dockerfile = config.get('dockerfile', 'Dockerfile')

    this.credentials = config.get('credentials', 'jenkins-dockerhub')

    this.mainBranch = config.get('mainBranch', 'master')
```

Reply:
Thanks for pointing this language improvement! I was not aware of this method : it is way more readable with your suggestion!